qiime2/sdk/__init__.py

```diff
@@ -9,14 +9,14 @@
 from .context import Context
 from .action import Action, Method, Visualizer, Pipeline
 from .plugin_manager import PluginManager, UninitializedPluginManagerError
-from .result import Result, Artifact, Visualization
```

Comment:
We should hoist `ResultCollection` up to the root `__init__` as it's perfectly reasonable for a Python API user to instantiate one of these, although kind of more work than they need to do. They might also want to do `isinstance` checks on `Results<>` which would be fine as well.